GarbageCollectionMetricSetTest unknownCount test failure

- Fails on `Hazelcast-3.x-IbmJDK8`
- Fails on [Build #60 (Jan 26, 2020 8:07:03 AM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.x-IbmJDK8/60/testReport/junit/com.hazelcast.internal.metrics.metricsets/GarbageCollectionMetricSetTest/unknownCount/)
- Error 
```
expected:<130.0> but was:<0.0>
```
- Stack Trace
```
java.lang.AssertionError: expected:<130.0> but was:<0.0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:575)
	at org.junit.Assert.assertEquals(Assert.java:700)
	at com.hazelcast.internal.metrics.metricsets.GarbageCollectionMetricSetTest$5.run(GarbageCollectionMetricSetTest.java:110)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1458)
	at com.hazelcast.internal.metrics.metricsets.GarbageCollectionMetricSetTest.unknownCount(GarbageCollectionMetricSetTest.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:818)
```

```
Hiccups measured while running test 'unknownCount(com.hazelcast.internal.metrics.metricsets.GarbageCollectionMetricSetTest):'
08:28:25, accumulated pauses: 52 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:28:30, accumulated pauses: 47 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:28:35, accumulated pauses: 42 ms, max pause: 1 ms, pauses over 1000 ms: 0
08:28:40, accumulated pauses: 40 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:28:45, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:28:50, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:28:55, accumulated pauses: 52 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:00, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:05, accumulated pauses: 56 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:10, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:15, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:20, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:25, accumulated pauses: 52 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:30, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:35, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:40, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:45, accumulated pauses: 53 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:50, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:29:55, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:30:00, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:30:05, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:30:10, accumulated pauses: 39 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:30:15, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
08:30:20, accumulated pauses: 52 ms, max pause: 1 ms, pauses over 1000 ms: 0
08:30:25, accumulated pauses: 2 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

kindly check

--- a/hazelcast/src/main/java/com/hazelcast/internal/metrics/MetricDescriptorConstants.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/metrics/MetricDescriptorConstants.java
@@ -166,7 +166,7 @@ public final class MetricDescriptorConstants {
     public static final String GC_METRIC_MINOR_TIME = "minorTime";
     public static final String GC_METRIC_MAJOR_COUNT = "majorCount";
     public static final String GC_METRIC_MAJOR_TIME = "majorTime";
-    public static final String GC_METRIC_UNKNOWN_COUNT = "unknowsnCount";
+    public static final String GC_METRIC_UNKNOWN_COUNT = "unknownCount";
     public static final String GC_METRIC_UNKNOWN_TIME = "unknownTime";
     // ===[/GC]=========================================================
 

--- a/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/ClassLoadingMetricSetTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/ClassLoadingMetricSetTest.java
@@ -22,6 +22,7 @@ import com.hazelcast.logging.Logger;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -48,6 +49,11 @@ public class ClassLoadingMetricSetTest extends HazelcastTestSupport {
     public void setup() {
         metricsRegistry = new MetricsRegistryImpl(Logger.getLogger(MetricsRegistryImpl.class), INFO);
         ClassLoadingMetricSet.register(metricsRegistry);
+    }
+
+    @After
+    public void tearDown() {
+        metricsRegistry.shutdown();
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/FileMetricSetTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/FileMetricSetTest.java
@@ -22,6 +22,7 @@ import com.hazelcast.logging.Logger;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -41,6 +42,11 @@ public class FileMetricSetTest extends HazelcastTestSupport {
     public void setup() {
         metricsRegistry = new MetricsRegistryImpl(Logger.getLogger(MetricsRegistryImpl.class), INFO);
         FileMetricSet.register(metricsRegistry);
+    }
+
+    @After
+    public void tearDown() {
+        metricsRegistry.shutdown();
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/GarbageCollectionMetricSetTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/GarbageCollectionMetricSetTest.java
@@ -19,10 +19,10 @@ package com.hazelcast.internal.metrics.metricsets;
 import com.hazelcast.internal.metrics.LongGauge;
 import com.hazelcast.internal.metrics.impl.MetricsRegistryImpl;
 import com.hazelcast.logging.Logger;
-import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -46,6 +46,11 @@ public class GarbageCollectionMetricSetTest extends HazelcastTestSupport {
         gcStats = new GarbageCollectionMetricSet.GcStats();
     }
 
+    @After
+    public void tearDown() {
+        metricsRegistry.shutdown();
+    }
+
     @Test
     public void utilityConstructor() {
         assertUtilityConstructor(GarbageCollectionMetricSet.class);
@@ -54,48 +59,36 @@ public class GarbageCollectionMetricSetTest extends HazelcastTestSupport {
     @Test
     public void minorCount() {
         final LongGauge gauge = metricsRegistry.newLongGauge("gc.minorCount");
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                gcStats.run();
-                assertEquals(gcStats.minorCount, gauge.read(), 1);
-            }
+        assertTrueEventually(() -> {
+            gcStats.run();
+            assertEquals(gcStats.minorCount, gauge.read(), 1);
         });
     }
 
     @Test
-    public void minorTime() throws InterruptedException {
+    public void minorTime() {
         final LongGauge gauge = metricsRegistry.newLongGauge("gc.minorTime");
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                gcStats.run();
-                assertEquals(gcStats.minorTime, gauge.read(), SECONDS.toMillis(1));
-            }
+        assertTrueEventually(() -> {
+            gcStats.run();
+            assertEquals(gcStats.minorTime, gauge.read(), SECONDS.toMillis(1));
         });
     }
 
     @Test
     public void majorCount() {
         final LongGauge gauge = metricsRegistry.newLongGauge("gc.majorCount");
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                gcStats.run();
-                assertEquals(gcStats.majorCount, gauge.read(), 1);
-            }
+        assertTrueEventually(() -> {
+            gcStats.run();
+            assertEquals(gcStats.majorCount, gauge.read(), 1);
         });
     }
 
     @Test
     public void majorTime() {
         final LongGauge gauge = metricsRegistry.newLongGauge("gc.majorTime");
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                gcStats.run();
-                assertEquals(gcStats.majorTime, gauge.read(), SECONDS.toMillis(1));
-            }
+        assertTrueEventually(() -> {
+            gcStats.run();
+            assertEquals(gcStats.majorTime, gauge.read(), SECONDS.toMillis(1));
         });
     }
 
@@ -103,24 +96,18 @@ public class GarbageCollectionMetricSetTest extends HazelcastTestSupport {
     @Test
     public void unknownCount() {
         final LongGauge gauge = metricsRegistry.newLongGauge("gc.unknownCount");
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                gcStats.run();
-                assertEquals(gcStats.unknownCount, gauge.read(), 1);
-            }
+        assertTrueEventually(() -> {
+            gcStats.run();
+            assertEquals(gcStats.unknownCount, gauge.read(), 1);
         });
     }
 
     @Test
     public void unknownTime() {
         final LongGauge gauge = metricsRegistry.newLongGauge("gc.unknownTime");
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                gcStats.run();
-                assertEquals(gcStats.unknownTime, gauge.read(), SECONDS.toMillis(1));
-            }
+        assertTrueEventually(() -> {
+            gcStats.run();
+            assertEquals(gcStats.unknownTime, gauge.read(), SECONDS.toMillis(1));
         });
     }
 }

--- a/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/OperatingSystemMetricSetTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/OperatingSystemMetricSetTest.java
@@ -24,6 +24,7 @@ import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.AssumptionViolatedException;
 import org.junit.Before;
 import org.junit.Test;
@@ -52,6 +53,11 @@ public class OperatingSystemMetricSetTest extends HazelcastTestSupport {
     public void setup() {
         metricsRegistry = new MetricsRegistryImpl(getLogger(MetricsRegistryImpl.class), INFO);
         OperatingSystemMetricSet.register(metricsRegistry);
+    }
+
+    @After
+    public void tearDown() {
+        metricsRegistry.shutdown();
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/RuntimeMetricSetTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/RuntimeMetricSetTest.java
@@ -23,6 +23,7 @@ import com.hazelcast.logging.Logger;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -47,6 +48,11 @@ public class RuntimeMetricSetTest extends HazelcastTestSupport {
         metricsRegistry = new MetricsRegistryImpl(Logger.getLogger(MetricsRegistryImpl.class), ProbeLevel.INFO);
         RuntimeMetricSet.register(metricsRegistry);
         runtime = Runtime.getRuntime();
+    }
+
+    @After
+    public void tearDown() {
+        metricsRegistry.shutdown();
     }
 
     @Test

--- a/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/ThreadMetricSetTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/metrics/metricsets/ThreadMetricSetTest.java
@@ -22,6 +22,7 @@ import com.hazelcast.logging.Logger;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -45,6 +46,11 @@ public class ThreadMetricSetTest extends HazelcastTestSupport {
     public void setup() {
         metricsRegistry = new MetricsRegistryImpl(Logger.getLogger(MetricsRegistryImpl.class), INFO);
         ThreadMetricSet.register(metricsRegistry);
+    }
+
+    @After
+    public void tearDown() {
+        metricsRegistry.shutdown();
     }
 
     @Test

